front/app/containers/ProjectsShowPage/timeline/index.tsx

```diff
@@ -148,18 +141,20 @@ const ProjectTimelineContainer = memo<Props & WithRouterProps>(
       // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [phases]);
 
+    const handleSetSelectedPhase = (phase: IPhaseData) => {
+      setSelectedPhase(phase);
+    };
+
     if (
       !isNilOrError(project) &&
       !isNilOrError(phases) &&
       phases.length > 0 &&
-      selectedPhase !== undefined
+      selectedPhase
     ) {
-      const selectedPhaseId = selectedPhase ? selectedPhase.id : null;
+      const selectedPhaseId = selectedPhase.id;
       const isPBPhase =
-        selectedPhase?.attributes?.participation_method === 'budgeting';
-      const participationMethod = !isNilOrError(selectedPhase)
-        ? selectedPhase.attributes.participation_method
-        : null;
+        selectedPhase.attributes.participation_method === 'budgeting';
+      const participationMethod = selectedPhase.attributes.participation_method;
```

Comment:
Simplified because selectedPhase can't be undefined + we shouldn't render the timeline if there is no selectedPhase (phase).